File: backend/src/main/java/com/leadgrowth/entity/FollowupReminder.java
package com.leadgrowth.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "followup_reminders")
public class FollowupReminder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lead_id", nullable = false)
    private Lead lead;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_to_id", nullable = false)
    private User assignedTo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "workspace_id", nullable = false)
    private Workspace workspace;

    @Column(nullable = false)
    private LocalDateTime scheduledAt;

    @Column(nullable = false)
    private String status; // UPCOMING, PENDING, COMPLETED, MISSED

    private String type; // CALL, EMAIL, MEETING, DEMO

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(columnDefinition = "TEXT")
    private String remarks;

    private String outcome;

    private LocalDateTime nextFollowupDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_id")
    private User createdBy;

    private LocalDateTime completedAt;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    public FollowupReminder() {}

    public FollowupReminder(Lead lead, User assignedTo, Workspace workspace, LocalDateTime scheduledAt, String type, String notes) {
        this.lead = lead;
        this.assignedTo = assignedTo;
        this.workspace = workspace;
        this.scheduledAt = scheduledAt;
        this.status = "UPCOMING";
        this.type = type != null ? type : "CALL";
        this.notes = notes;
        this.remarks = notes;
        this.createdAt = LocalDateTime.now();
    }

    public FollowupReminder(Lead lead, User assignedTo, Workspace workspace, LocalDateTime scheduledAt, String type, String notes, String outcome, LocalDateTime nextFollowupDate, User createdBy) {
        this.lead = lead;
        this.assignedTo = assignedTo;
        this.workspace = workspace;
        this.scheduledAt = scheduledAt;
        this.status = "UPCOMING";
        this.type = type != null ? type : "CALL";
        this.notes = notes;
        this.remarks = notes;
        this.outcome = outcome;
        this.nextFollowupDate = nextFollowupDate;
        this.createdBy = createdBy;
        this.createdAt = LocalDateTime.now();
    }

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (status == null) {
            status = "UPCOMING";
        }
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Lead getLead() { return lead; }
    public void setLead(Lead lead) { this.lead = lead; }

    public User getAssignedTo() { return assignedTo; }
    public void setAssignedTo(User assignedTo) { this.assignedTo = assignedTo; }

    public Workspace getWorkspace() { return workspace; }
    public void setWorkspace(Workspace workspace) { this.workspace = workspace; }

    public LocalDateTime getScheduledAt() { return scheduledAt; }
    public void setScheduledAt(LocalDateTime scheduledAt) { this.scheduledAt = scheduledAt; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }

    public String getRemarks() { return remarks; }
    public void setRemarks(String remarks) { this.remarks = remarks; }

    public String getOutcome() { return outcome; }
    public void setOutcome(String outcome) { this.outcome = outcome; }

    public LocalDateTime getNextFollowupDate() { return nextFollowupDate; }
    public void setNextFollowupDate(LocalDateTime nextFollowupDate) { this.nextFollowupDate = nextFollowupDate; }

    public User getCreatedBy() { return createdBy; }
    public void setCreatedBy(User createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getCompletedAt() { return completedAt; }
    public void setCompletedAt(LocalDateTime completedAt) { this.completedAt = completedAt; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: backend-dotnet/Models/FollowupReminder.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LeadGrowth.Models;

[Table("followup_reminders")]
public class FollowupReminder
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    [Column("workspace_id")]
    public long WorkspaceId { get; set; }

    [ForeignKey("WorkspaceId")]
    public virtual Workspace Workspace { get; set; } = null!;

    [Required]
    [Column("lead_id")]
    public long LeadId { get; set; }

    [ForeignKey("LeadId")]
    public virtual Lead Lead { get; set; } = null!;

    [Column("assigned_to_id")]
    public long? AssignedToId { get; set; }

    [ForeignKey("AssignedToId")]
    public virtual User? AssignedTo { get; set; }

    [Required]
    [Column("scheduled_at")]
    public DateTime ScheduledAt { get; set; }

    [Required]
    [MaxLength(30)]
    public string Type { get; set; } = "CALL";

    [Column(TypeName = "TEXT")]
    public string? Notes { get; set; }

    [Required]
    [MaxLength(20)]
    public string Status { get; set; } = "UPCOMING";

    [NotMapped]
    public bool ConflictFlag { get; set; } = false;

    [Column("outcome")]
    [MaxLength(50)]
    public string? Outcome { get; set; }

    [Column("remarks", TypeName = "TEXT")]
    public string? Remarks { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
